src/firstboot/firstboot.c

```diff
@@ -1390,41 +1413,52 @@ static int run(int argc, char *argv[]) {
                                 DISSECT_IMAGE_FSCK |
                                 DISSECT_IMAGE_GROWFS,
                                 &unlink_dir,
-                                NULL,
+                                &rfd,
                                 &loop_device);
                 if (r < 0)
                         return r;
 
                 arg_root = strdup(unlink_dir);
                 if (!arg_root)
                         return log_oom();
+        } else if (arg_root) {
+                rfd = open(arg_root, O_DIRECTORY|O_PATH|O_CLOEXEC);
+                if (rfd < 0)
+                        return log_error_errno(errno, "Failed to open %s: %m", arg_root);
+        } else
+                rfd = AT_FDCWD;
```

Comment:
hm? the current working directory? shouldn't this be `open("/")`?

Reply:
`AT_FDCWD` means either the current working directory or root, depending on whether the path is absolute or not. Of course I'm actually using relative paths, but that's wrong and it should be absolute paths. We'll also need #25960 to make this work properly.

Reply:
Turns out absolute paths are the wrong approach and relative paths are the way to go so disregard the earlier comment.